Sources/Atomic.swift

```diff
@@ -8,56 +8,107 @@
 
 import Foundation
 
-final class PosixThreadMutex: NSLocking {
-	private var mutex = pthread_mutex_t()
+internal struct _PosixThreadMutex {
+	private let mutex: UnsafeMutablePointer<pthread_mutex_t>
 
 	init() {
-		let result = pthread_mutex_init(&mutex, nil)
+		mutex = UnsafeMutablePointer<pthread_mutex_t>.allocate(capacity: 1)
+		let result = pthread_mutex_init(mutex, nil)
 		precondition(result == 0, "Failed to initialize mutex with error \(result).")
 	}
 
-	deinit {
-		let result = pthread_mutex_destroy(&mutex)
+	func deinitialize() {
+		let result = pthread_mutex_destroy(mutex)
 		precondition(result == 0, "Failed to destroy mutex with error \(result).")
+		mutex.deallocate(capacity: 1)
```

Comment:
https://developer.apple.com/reference/swift/unsafemutablepointer/2295090-deallocate:

> Precondition: The memory is not initialized.

This is what made me think that perhaps we should be calling `mutex.deinitialize()` first.